src/Controller/Component/SecurityComponent.php

```diff
@@ -102,7 +102,7 @@ public function startup(Event $event)
         $controller = $event->getSubject();
         $this->session = $controller->request->getSession();
         $this->_action = $controller->request->getParam('action');
-        $hasData = (bool)$controller->request->getData();
+        $hasData = ($controller->request->getData() || $controller->request->is(['put', 'post', 'delete', 'patch']));
```

Comment:
I didn't remove `$controller->request->getData()` as it was needed to pass existing tests.